static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.spec.ts

```diff
@@ -26,6 +36,60 @@ const createAd = (html: string) => {
 const getAd = (): HTMLElement =>
 	document.querySelector(adSelector) as HTMLElement;
 
+describe('shouldRenderLabel', () => {
+	afterEach(() => {
+		document.body.innerHTML = '';
+	});
+
+	it('renders an ad label for normal ads', async () => {
+		createAd(adverts['withLabel']);
+		return renderAdvertLabel(getAd()).then(() => {
+			const ad = getAd();
+			expect(shouldRenderLabel(ad)).toBeTruthy();
+		});
+	});
+
+	it('renders an ad label for interscroller ads', async () => {
+		createAd(adverts['interscroller']);
+		return renderAdvertLabel(getAd()).then(() => {
+			const ad = getAd();
+			expect(shouldRenderLabel(ad)).toBeTruthy();
+		});
+	});
+
+	it('does NOT render an ad label for fluid ads', async () => {
+		createAd(adverts['fluid']);
+		return renderAdvertLabel(getAd()).then(() => {
+			const ad = getAd();
+			expect(shouldRenderLabel(ad)).toBeFalsy();
+		});
+	});
+
+	it('does NOT render an ad label for collapsed ads', async () => {
+		createAd(adverts['collapse']);
+		return renderAdvertLabel(getAd()).then(() => {
+			const ad = getAd();
+			expect(shouldRenderLabel(ad)).toBeFalsy();
+		});
+	});
+
+	it('renders an ad label when data label attribute is true', async () => {
+		createAd(adverts['dataLabelTrue']);
+		return renderAdvertLabel(getAd()).then(() => {
+			const ad = getAd();
+			expect(shouldRenderLabel(ad)).toBeTruthy();
+		});
+	});
+
+	it('does NOT render an ad label when data label attribute is false', async () => {
```

Comment:
This is basically a repeat of a test further down in the file (line 111). Maybe it makes sense to delete the old tests and just have these new ones as they're more thorough. We can only test shouldRenderLabel anyway! (see comment on line 98).

Reply:
Ah thanks, good spot